### PR TITLE
`azurerm_spring_cloud_gateway` - support for the `allowed_origin_patterns` property

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_gateway_resource.go [hunk 1]
```diff
@@ -58,6 +59,13 @@ func resourceSpringCloudGateway() *pluginsdk.Resource {
 				ValidateFunc: validate.SpringCloudServiceID,
 			},
 
+			"addon_json": {
```
Comment (jackofallops) [inline internal/services/springcloud/spring_cloud_gateway_resource.go:62]: I think we should be explicit here for the currently supported configuration, not arbitrary JSON. The fact that it is noted to be subject to change in the docs in future is not really any different to any other API parameter. We have known supported values today, and have the ability to update/deprecate etc in future as with any other property or set of properties.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @ms-henglu 
Thanks for this PR.  I think we need to be explicit in the values for this new property and create an appropriate schema block for it.  Please see the comment below and we can take another look when that is addressed.

Thanks!

File: internal/services/springcloud/spring_cloud_gateway_resource.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_gateway_resource.go]: Got it, I've removed this `addon_json` field. Please take another look, thanks!

**Review by katbyte — APPROVED** [review]
LGTM 💾

**Pull request merged**